fedbadges/utils.py

```diff
@@ -111,13 +112,25 @@ def notification_callback(topic, msg):
 def user_exists_in_fas(config, user):
     """ Return true if the user exists in FAS. """
     default_url = 'https://admin.fedoraproject.org/accounts/'
+    if config.get["freeipa-fas"]:
+        return user_exists_in_freeipa_fas(config, user)
+
     fas2 = fedora.client.AccountSystem(
         base_url=config['fas_credentials'].get('base_url', default_url),
         username=config['fas_credentials']['username'],
         password=config['fas_credentials']['password'],
     )
     return bool(fas2.person_by_username(user))
 
+def user_exists_in_freeipa_fas(config, user):
+    """ Return true if the user exists in freeipa-fas. """
+    default_url = 'https://admin.fedoraproject.org/accounts/'
```

Comment:
It feels like this default URL should be a constant in `fasjson_client`, no?

Reply:
yep, good point. I'll send a PR and remove it from here.

Reply:
Actually I'm not so sure about that, there are pros and cons that we need to think about.

Reply:
`python-fedora` sets it as a default in `__init__`. I wasn't thinking constant, more a default if you don't provide it. A pro is that the client becomes even easier to use, a con maybe that we are coupling a little more tightly?

Reply:
Either we put that URL in the app's configuration file, and it will be different in staging and prod, or we put it in fasjson-client's config file, which was only used for CLI as of now, and we end up having to deploy an additional config file with the app only for that value. Now I'm thinking of the former option, keep it a config value in the app's own config file.

Which would have to be done here too because the URL is going to be different in staging, so we can't hardcode the value.

Reply:
From a deployment point of view, the value will have to be set in the configuration file so that Ansible can put a different value in staging and prod, so we won't be able to rely on the default anyway.

Reply:
OK as it stands then having the default value here of `https://admin.fedoraproject.org/accounts/` actually does exactly what you describe because the apps config checks for staging and provides a different URL.

Reply:
Well, I'd say it depends on who's gonna maintain this code :man_shrugging: . It'd be best to have a global default value  at the top to update this easier.